test/Feature/HLSLLib/ldexp.32.test

```diff
@@ -0,0 +1,62 @@
+#--- source.hlsl
+
+StructuredBuffer<float4> In : register(t0);
+RWStructuredBuffer<float4> Out : register(u1);
+
+[numthreads(1,1,1)]
+void main() {
+  Out[0] = ldexp(In[0], In[1]);
+  Out[1].x = ldexp(In[1].x, In[1].y);
+  Out[1].yzw = ldexp(In[1].yzw, In[0].yzw);
+  Out[2].xy = ldexp(In[0].xy, In[1].yw);
+}
+
+//--- pipeline.yaml
+
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
+Buffers:
+  - Name: In
+    Format: Float32
+    Stride: 16
+    Data: [ 3.14159, 0, -inf, NaN, 1, 1.5, -0.5, inf ]
```

Comment:
The HLSL docs on this function are light on details, but referencing the c++ docs: https://en.cppreference.com/w/cpp/numeric/math/ldexp, I think it would be good to test overflow and underflow situations as well. 

Looking at the special cases it seems the only one missing, besides overflow and underflow, is where the first argument (num) is -0.